packages/block-editor/src/components/block-list/block.native.js

```diff
@@ -188,6 +189,17 @@ class BlockListBlock extends Component {
 		];
 	}
 
+	applyToolbarStyle() {
+		const {
+			hasChildren,
+			isUnregisteredBlock,
+		} = this.props;
+
+		if ( ! hasChildren || isUnregisteredBlock ) {
```

Comment:
I think you don't need the extra `isUnregisteredBlock` check here, since that's included as part of `hasChildren` with the latest changes

Reply:
Sure, it's a leftover. Thanks for a catch!